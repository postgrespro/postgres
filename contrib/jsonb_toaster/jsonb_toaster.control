# jsonb_toaster extension
comment = 'jsonb_toaster - updatable jsonb toaster'
default_version = '1.0'
module_pathname = '$libdir/jsonb_toaster'
relocatable = true
