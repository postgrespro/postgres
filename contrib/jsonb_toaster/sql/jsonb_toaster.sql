CREATE EXTENSION jsonb_toaster;

CREATE TABLE tst_failed (
	t json TOASTER jsonb_toaster
);

CREATE TABLE tst1 (
	t jsonb TOASTER jsonb_toaster
);

CREATE TABLE tst2 (
	t jsonb
);

ALTER TABLE tst2 ALTER COLUMN t SET TOASTER jsonb_toaster;


CREATE TABLE test_jsonb_toaster (id int, jb jsonb);
ALTER TABLE test_jsonb_toaster ALTER jb SET TOASTER jsonb_toaster;

INSERT INTO test_jsonb_toaster
SELECT i, (
	SELECT jsonb_object_agg('key' || j, jsonb_build_array(repeat('a', pow(2, i + j)::int)))
	FROM generate_series(1,10) j
)
FROM generate_series(1, 10) i;

SELECT id, key, pg_column_size(value::text) FROM test_jsonb_toaster, jsonb_each(jb);

DROP TABLE test_jsonb_toaster;


create table test_jsonbz_arr (id int, js jsonb toaster jsonb_toaster);

insert into test_jsonbz_arr
select
  j id,
  jsonb_build_object(
     'a', jsonb_agg(repeat('a', pow(2, 6 + i)::int)),
     'b', 'foo',
     'c', jsonb_agg(jsonb_build_object('a', repeat('a', pow(2, 6 + i)::int), 'b', 1))
  ) js
from
  generate_series(0, 19) j,
  generate_series(0, j) i
group by j
order by j;

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('b', 64)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('c', 64)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('d', 65)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('e', 65)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;


update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('b', 64)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('c', 64)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('d', 65)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('e', 65)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;
