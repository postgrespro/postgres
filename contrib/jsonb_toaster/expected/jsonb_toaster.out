CREATE EXTENSION jsonb_toaster;
CREATE TABLE tst_failed (
	t json TOASTER jsonb_toaster
);
CREATE TABLE tst1 (
	t jsonb TOASTER jsonb_toaster
);
CREATE TABLE tst2 (
	t jsonb
);
ALTER TABLE tst2 ALTER COLUMN t SET TOASTER jsonb_toaster;
CREATE TABLE test_jsonb_toaster (id int, jb jsonb);
ALTER TABLE test_jsonb_toaster ALTER jb SET TOASTER jsonb_toaster;
INSERT INTO test_jsonb_toaster
SELECT i, (
	SELECT jsonb_object_agg('key' || j, jsonb_build_array(repeat('a', pow(2, i + j)::int)))
	FROM generate_series(1,10) j
)
FROM generate_series(1, 10) i;
SELECT id, key, pg_column_size(value::text) FROM test_jsonb_toaster, jsonb_each(jb);
 id |  key  | pg_column_size 
----+-------+----------------
  1 | key1  |             12
  1 | key2  |             16
  1 | key3  |             24
  1 | key4  |             40
  1 | key5  |             72
  1 | key6  |            136
  1 | key7  |            264
  1 | key8  |            520
  1 | key9  |           1032
  1 | key10 |           2056
  2 | key1  |             16
  2 | key2  |             24
  2 | key3  |             40
  2 | key4  |             72
  2 | key5  |            136
  2 | key6  |            264
  2 | key7  |            520
  2 | key8  |           1032
  2 | key9  |           2056
  2 | key10 |           4104
  3 | key1  |             24
  3 | key2  |             40
  3 | key3  |             72
  3 | key4  |            136
  3 | key5  |            264
  3 | key6  |            520
  3 | key7  |           1032
  3 | key8  |           2056
  3 | key9  |           4104
  3 | key10 |           8200
  4 | key1  |             40
  4 | key2  |             72
  4 | key3  |            136
  4 | key4  |            264
  4 | key5  |            520
  4 | key6  |           1032
  4 | key7  |           2056
  4 | key8  |           4104
  4 | key9  |           8200
  4 | key10 |          16392
  5 | key1  |             72
  5 | key2  |            136
  5 | key3  |            264
  5 | key4  |            520
  5 | key5  |           1032
  5 | key6  |           2056
  5 | key7  |           4104
  5 | key8  |           8200
  5 | key9  |          16392
  5 | key10 |          32776
  6 | key1  |            136
  6 | key2  |            264
  6 | key3  |            520
  6 | key4  |           1032
  6 | key5  |           2056
  6 | key6  |           4104
  6 | key7  |           8200
  6 | key8  |          16392
  6 | key9  |          32776
  6 | key10 |          65544
  7 | key1  |            264
  7 | key2  |            520
  7 | key3  |           1032
  7 | key4  |           2056
  7 | key5  |           4104
  7 | key6  |           8200
  7 | key7  |          16392
  7 | key8  |          32776
  7 | key9  |          65544
  7 | key10 |         131080
  8 | key1  |            520
  8 | key2  |           1032
  8 | key3  |           2056
  8 | key4  |           4104
  8 | key5  |           8200
  8 | key6  |          16392
  8 | key7  |          32776
  8 | key8  |          65544
  8 | key9  |         131080
  8 | key10 |         262152
  9 | key1  |           1032
  9 | key2  |           2056
  9 | key3  |           4104
  9 | key4  |           8200
  9 | key5  |          16392
  9 | key6  |          32776
  9 | key7  |          65544
  9 | key8  |         131080
  9 | key9  |         262152
  9 | key10 |         524296
 10 | key1  |           2056
 10 | key2  |           4104
 10 | key3  |           8200
 10 | key4  |          16392
 10 | key5  |          32776
 10 | key6  |          65544
 10 | key7  |         131080
 10 | key8  |         262152
 10 | key9  |         524296
 10 | key10 |        1048584
(100 rows)

DROP TABLE test_jsonb_toaster;
create table test_jsonbz_arr (id int, js jsonb toaster jsonb_toaster);
insert into test_jsonbz_arr
select
  j id,
  jsonb_build_object(
     'a', jsonb_agg(repeat('a', pow(2, 6 + i)::int)),
     'b', 'foo',
     'c', jsonb_agg(jsonb_build_object('a', repeat('a', pow(2, 6 + i)::int), 'b', 1))
  ) js
from
  generate_series(0, 19) j,
  generate_series(0, j) i
group by j
order by j;
update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('b', 64)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;
 id |                             ?column?                             
----+------------------------------------------------------------------
  0 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  1 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  2 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  3 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  4 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  5 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  6 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  7 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  8 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  9 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 10 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 11 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 12 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 13 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 14 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 15 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 16 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 17 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 18 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 19 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('c', 64)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;
 id |                             ?column?                             
----+------------------------------------------------------------------
  0 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  1 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  2 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  3 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  4 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  5 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  6 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  7 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  8 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  9 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 10 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 11 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 12 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 13 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 14 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 15 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 16 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 17 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 18 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 19 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('d', 65)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;
 id |                             ?column?                              
----+-------------------------------------------------------------------
  0 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  1 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  2 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  3 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  4 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  5 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  6 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  7 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  8 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  9 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 10 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 11 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 12 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 13 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 14 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 15 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 16 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 17 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 18 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 19 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{a,0}', to_jsonb(repeat('e', 65)));
select id, js->'a'->>0 from test_jsonbz_arr order by id;
 id |                             ?column?                              
----+-------------------------------------------------------------------
  0 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  1 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  2 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  3 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  4 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  5 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  6 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  7 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  8 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  9 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 10 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 11 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 12 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 13 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 14 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 15 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 16 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 17 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 18 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 19 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('b', 64)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;
 id |                             ?column?                             
----+------------------------------------------------------------------
  0 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  1 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  2 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  3 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  4 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  5 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  6 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  7 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  8 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
  9 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 10 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 11 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 12 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 13 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 14 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 15 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 16 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 17 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 18 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 19 | bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('c', 64)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;
 id |                             ?column?                             
----+------------------------------------------------------------------
  0 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  1 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  2 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  3 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  4 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  5 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  6 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  7 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  8 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
  9 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 10 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 11 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 12 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 13 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 14 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 15 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 16 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 17 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 18 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 19 | cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('d', 65)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;
 id |                             ?column?                              
----+-------------------------------------------------------------------
  0 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  1 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  2 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  3 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  4 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  5 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  6 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  7 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  8 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
  9 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 10 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 11 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 12 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 13 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 14 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 15 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 16 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 17 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 18 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
 19 | ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd
(20 rows)

update test_jsonbz_arr set js = jsonb_set(js, '{c,0,a}', to_jsonb(repeat('e', 65)));
select id, js->'c'->0->>'a' from test_jsonbz_arr order by id;
 id |                             ?column?                              
----+-------------------------------------------------------------------
  0 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  1 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  2 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  3 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  4 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  5 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  6 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  7 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  8 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
  9 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 10 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 11 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 12 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 13 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 14 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 15 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 16 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 17 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 18 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
 19 | eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
(20 rows)

